homework/list.js

```diff
@@ -2,43 +2,51 @@
 
 (function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	function listObject(){
+		this._listNoteContent = [];
+		this._wrapper = document.querySelector('#note-list-wrapper');
+	}
+  
+  listObject.prototype.start = function() {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
+    this.fetchList()
+    .then(this.updateList.bind(this))
+    .then(this.drawList.bind(this))
+    
+    .then(this.preloadFirstNote.bind(this));
+   
+    
     window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+      this.onNoteOpen(event);
+    }.bind(this));
   }
 
-  function onNoteOpen(event) {
+  listObject.prototype.onNoteOpen = function(event) {
```

Comment:
It's a convention to prefix `on` to a event handler with it's corresponding event. So this function should be renamed as `onClick`.